# Hypothesis Testing for Weekly Operating Costs
**Bombay Hospitality Ltd**

In [1]:
import math
from scipy.stats import norm

In [2]:
# Given data
sample_mean = 3050
X_mean = 600  # mean units produced
X_std = 25  # std dev of units
n = 25

In [3]:
# Theoretical weekly cost model
# W = 1000 + 5X
w_mean = 1000 + 5 * X_mean

In [4]:
# Standard deviation of wee cost
w_std = 5 * X_std

 **Hypothesis:**

Null hypothesis (H0): The true mean weekly cost = w_mean (no increase).
Alternative (H1): The true mean weekly cost > w_mean (owners claim costs are higher).
This is a one-tailed (right) test.

In [5]:
# Test statistic
z = (sample_mean - w_mean) / (w_std / math.sqrt(n))

In [6]:
# Critical value for alpha = 0.05 
z_critical = norm.ppf(0.95)

In [7]:
# p‑value
tail_prob = 1 - norm.cdf(z)

print("Theoretical Mean Cost (mu_w):", w_mean)
print("Standard Deviation of Weekly Cost (w_std):", w_std)
print("Z‑Statistic:", z)
print("Critical Value (z_critical):", z_critical)
print("p‑value:", tail_prob)

Theoretical Mean Cost (mu_w): 4000
Standard Deviation of Weekly Cost (w_std): 125
Z‑Statistic: -38.0
Critical Value (z_critical): 1.6448536269514722
p‑value: 1.0


In [8]:
# Decision
if z > z_critical:
    print("Decision: Reject null hypothesis H0.")
    print("Conclusion: There is statistically significant evidence (at alpha = 0.05) that ")
    print("the true mean weekly operating cost is higher than the theoretical model suggests.")
    print(" the true mean weekly operating cost is higher than the theoretical model suggests.")
else:
    print("Decision: Fail to Reject null hypothesis (H0).")
    print("Conclusion: There is not enough statistical evidence (at alpha = 0.05) to conclude ")
    print("that the true mean weekly operating cost is higher than the theoretical model suggests.")
    print("that the true mean weekly operating cost is higher than the theoretical model suggests.")

Decision: Fail to Reject null hypothesis (H0).
Conclusion: There is not enough statistical evidence (at alpha = 0.05) to conclude 
that the true mean weekly operating cost is higher than the theoretical model suggests.
that the true mean weekly operating cost is higher than the theoretical model suggests.


# ESTIMATION AND CONFIDENCE INTERVALS

**99% Confidence Interval Using Sample Standard Deviation**


In [9]:
import numpy as np
from scipy import stats

In [10]:
# Data
data = np.array([1.13, 1.55, 1.43, 0.92, 1.25, 1.36, 1.32, 0.85, 1.07, 1.48,
                 1.20, 1.33, 1.18, 1.22, 1.29])

In [16]:
# 99% CI using sample standard deviation
n = len(data)
mean = np.mean(data)
s = np.std(data, ddof=1)

alpha = 0.01  # for 99% CI
df = n - 1    # degrees of freedom

In [17]:
# t critical value
t_critical = stats.t.ppf(1 - alpha/2, df)

In [18]:
# Standard error
SE = s / np.sqrt(n)

In [19]:
# Margin of error
ME_t = t_critical * SE

In [25]:
# Confidence interval
CI_t_lower = mean - ME_t
CI_t_upper = mean + ME_t

print("99% CI using sample Standard Deviation:")
print(f"Mean = {mean:.4f}, SD = {s:.4f}")
print(f"CI = ({CI_t_lower:.4f}, {CI_t_upper:.4f})\n")


99% CI using sample Standard Deviation:
Mean = 1.2387, SD = 0.1932
CI = (1.0902, 1.3871)



We are 99% confident the true mean durability lies between 1.090 and 1.387 million characters, given the sample and unknown σ.

**99% Confidence Interval Using Known Population Standard Deviation**

In [21]:
# 99% CI with KNOWN population SD

sigma = 0.2  # given population SD

# z critical value for 99% CI
z_critical = stats.norm.ppf(1 - alpha/2)

In [22]:
# Standard error with known sigma
SE_known = sigma / np.sqrt(n)

In [23]:
# Margin of error
ME_z = z_critical * SE_known

In [26]:
# Confidence interval
CI_z_lower = mean - ME_z
CI_z_upper = mean + ME_z

print("99% CI using KNOWN population Standard Deviation:")
print(f"CI = ({CI_z_lower:.4f}, {CI_z_upper:.4f})")


99% CI using KNOWN population Standard Deviation:
CI = (1.1057, 1.3717)


With σ = 0.2 assumed known, we are 99% confident the population mean lies between 1.105 and 1.371 million